simulator/src/replay/cmd_alloc_dump.rs

```diff
@@ -0,0 +1,220 @@
+use super::ledger_transaction_execution::*;
+use super::txn_reader::TxnReader;
+use super::Error;
+use crate::replay::ledger_transaction::PreparedLedgerTransactionInner;
+use clap::Parser;
+use flate2::read::GzDecoder;
+use flume;
+use radix_engine::types::*;
+use radix_engine::vm::wasm::*;
+use radix_engine::vm::ScryptoVm;
+use radix_engine_interface::prelude::NetworkDefinition;
+use radix_engine_profiling::info_alloc::*;
+use radix_engine_store_interface::db_key_mapper::SpreadPrefixKeyMapper;
+use radix_engine_store_interface::interface::CommittableSubstateDatabase;
+use radix_engine_stores::rocks_db_with_merkle_tree::RocksDBWithMerkleTreeSubstateStore;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::prelude::*;
+use std::path::PathBuf;
+use std::thread;
+use std::time::Duration;
+use tar::Archive;
+use transaction::prelude::HasSystemTransactionHash;
+use transaction::prelude::IntentHash;
+use transaction::prelude::TransactionHashBech32Encoder;
+
+/// Run transactions in archive using RocksDB and dump memory allocations
+#[derive(Parser, Debug)]
+pub struct TxnAllocDump {
+    /// Path to the source Node state manager database
+    pub source: PathBuf,
+    /// Path to a folder for storing state
+    pub database_dir: PathBuf,
+    /// Path to the output file
+    pub output_file: PathBuf,
+
+    /// The network to use, [mainnet | stokenet]
+    #[clap(short, long)]
+    pub network: Option<String>,
+    /// The max version to execute
+    #[clap(short, long)]
+    pub max_version: Option<u64>,
+
+    /// Ignore dump of user type of transactions
+    #[clap(long)]
+    pub ignore_dump_user_tx: bool,
+    /// Dump genesis type of transactions
+    #[clap(short = 'g', long)]
+    pub dump_genesis_tx: bool,
+    /// Dump round update type of transactions
+    #[clap(short = 'r', long)]
+    pub dump_round_update_tx: bool,
```

Comment:
Suggest rename to `--exclude-user-transaction`, `--include-generic-transaction` and `--include-round-update-transaction`